lib/ecto/changeset.ex

```diff
@@ -1576,9 +1580,13 @@ defmodule Ecto.Changeset do
 
     errors =
       case Map.fetch(changeset.params, error_param) do
-        {:ok, ^value} -> []
-        {:ok, _}      -> [{error_field, {message(opts, "does not match confirmation"), []}}]
-        :error        -> confirmation_missing(opts, error_field)
+        {:ok, ^value} ->
+          []
+        {:ok, _} ->
+          [{error_field,
+           {message(opts, "does not match confirmation"), [validation: :confirmation_doesnt_match]}}]
```

Comment:
I think this validation should be simply `:confirmation` and the one below should be `:required`?